code/datums/elements/corrupted_organ.dm

```diff
@@ -0,0 +1,66 @@
+/// Component applying shared behaviour by cursed organs granted when sacrificed by a heretic
+/// Mostly just does something spooky when it is removed
+/datum/element/corrupted_organ
+
+/datum/element/corrupted_organ/Attach(datum/target)
+	. = ..()
+	if (!isinternalorgan(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ORGAN_SURGICALLY_REMOVED, PROC_REF(on_removed))
+
+	var/atom/atom_parent = target
+	atom_parent.color = COLOR_VOID_PURPLE
+
+	atom_parent.add_filter(name = "ray", priority = 1, params = list(
+		type = "rays",
+		size = 12,
+		color = COLOR_VOID_PURPLE,
+		density = 12
+	))
+	var/ray_filter = atom_parent.get_filter("ray")
+	animate(ray_filter, offset = 100, time = 2 MINUTES, loop = -1, flags = ANIMATION_PARALLEL) // Absurdly long animate so nobody notices it hitching when it loops
+	animate(offset = 0, time = 2 MINUTES) // I sure hope duration of animate doesnt have any performance effect
+
+/datum/element/corrupted_organ/Detach(datum/source)
+	UnregisterSignal(source, list(COMSIG_ORGAN_SURGICALLY_REMOVED))
+	return ..()
+
+/// When we're taken out of someone, do something spooky
+/datum/element/corrupted_organ/proc/on_removed(atom/organ, mob/living/carbon/loser)
+	SIGNAL_HANDLER
+	if (loser.has_reagent(/datum/reagent/water/holywater) || HAS_TRAIT(loser, TRAIT_ANTIMAGIC) || prob(20))
```

Comment:
Fairly certain you want to check `can_block_magic` rather than `TRAIT_ANTIMAGIC`